sound/soc/intel/boards/sof_sdw_rt711.c

```diff
@@ -133,6 +133,20 @@ static int rt711_rtd_init(struct snd_soc_pcm_runtime *rtd)
 	return ret;
 }
 
+int sof_sdw_rt711_exit(struct device *dev, struct snd_soc_dai_link *dai_link)
+{
+	struct device *sdw_dev;
+
+	sdw_dev = bus_find_device_by_name(&sdw_bus_type, NULL,
+					  dai_link->codecs[0].name);
+	if (!sdw_dev)
+		return -EINVAL;
+
+	device_remove_properties(sdw_dev);
```

Comment:
missing put_device(sdw_dev); 
see init() code for reference.

Reply:
Fixed. @plbossart Could you teach me when should we call put_device()?

Reply:
@bardliao I am not the expert here, I just copy/paste :-) In this case the bus_get_device takes a reference so you have to release it to avoid issues with usage counts

Reply:
> @bardliao I am not the expert here, I just copy/paste :-) In this case the bus_get_device takes a reference so you have to release it to avoid issues with usage counts

Thanks @plbossart I was looking for get_device() from device_remove_properties(). Now I see it is from bus_find_device_by_name() -> bus_find_device() -> get_device().